Sources/Flash/Middlewares/FlashMiddleware.swift

```diff
@@ -1,43 +1,23 @@
 import Vapor
 
-public struct FlashMiddleware: Middleware, ServiceType {
-    private static let sessionKey = "_flash"
+/// Middleware that passes along any flashes from the session to the next incoming request.
+public struct FlashMiddleware: Middleware {
+    public func respond(to request: Request, chainingTo next: Responder) -> EventLoopFuture<Response> {
+        handle(request: request)
 
-    private let jsonEncoder = JSONEncoder()
-    private let jsonDecoder = JSONDecoder()
+        return next.respond(to: request).map { response in
+            self.handle(request: request, response: response)
 
-    /// See `ServiceType`.
-    public static func makeService(for container: Container) throws -> FlashMiddleware {
-        return .init()
+            return response
+        }
     }
 
-    public init() {}
-
-    /// See `Middleware`.
-    public func respond(to req: Request, chainingTo next: Responder) throws -> Future<Response> {
-        try handle(req: req)
-        return try next.respond(to: req)
-            .map(to: Response.self) { resp in
-                try self.handle(req: req, resp: resp)
-                return resp
-            }
-    }
-
-    private func handle(req: Request) throws {
-        guard let data = try req.session()[FlashMiddleware.sessionKey]?.data(using: .utf8) else { return }
-
-        let flash = try jsonDecoder.decode(FlashContainer.self, from: data)
-        let container = try req.privateContainer.make(FlashContainer.self)
-        container.new = flash.new
-        container.old = flash.old
+    private func handle(request: Request) {
+        request.flashes.removeAll()
```

Comment:
What if a previous middleware wants to add some in-flight flashes?

Reply:
The original logic here seemed to overwrite the existing in-flight flashes each time the middleware was called. I think what is implemented here is consistent with the previous version… although, I'm happy to be proven wrong! There's no documentation on the expected interactions between the various components here.

Reply:
I've had a look, and I was mis-handling this code — it looks like the logic is:

1. Copy any flashes from the session into the request
2. Respond to the request
3. Copy any flashes from the response into the session

I've adjusted the logic to reflect that now. I've also made it possible to accumulate flashes from layered middleware.